core/Network.js

```diff
@@ -77,16 +77,12 @@ Network.prototype.getProcessByName = function (processName) {
 
 Network.prototype.run = function (runtime, options, callback) {
   options = options || {};
-  function setPortRuntime(port) {
-    port[1].setRuntime(runtime);
-  }
 
   _.forEach(this._processes, function (process) {
-    process.inports.forEach(setPortRuntime);
-    process.outports.forEach(setPortRuntime);
+    _.invokeMap(process.inports, 'setRuntime', runtime);
```

Comment:
_invokeMap just loops over a collection of object, calling the method given by the string with the arguments that follow.